packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -2164,6 +2165,7 @@ class AsyncImage extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return new RawImageResource(
+      key: new ObjectKey(provider),
```

Comment:
Or, even better, just use the same mechanism as with NetworkImage, and key off the ImageResource.